lib/src/list.dart

```diff
@@ -165,6 +167,9 @@ class ManagedRealmList<T extends Object?> with RealmEntity, ListMixin<T> impleme
     final frozenRealm = realm.freeze();
     return frozenRealm.resolveList(this)!;
   }
+
+  @override
+  RealmResults<T> get asResults => RealmResultsInternal.createFromList(this, realm, _metadata);
```

Comment:
There are no tests for this API when used on a list of objects. I think we should add some to make sure it behaves as expected when it comes to querying and sorting.

Reply:
Yeah, this is testing it for results of primitives - I was thinking that we need a test for results of objects - something like:

```dart
final results = myList.asResults.query('foo > 5 SORT(bar)');

// assert on the order and contents

realm.deleteMany(results);
// assert that only the results matches were deleted and not everything from the list
```